partner_communication_switzerland/wizards/portal_wizard.py

```diff
@@ -9,25 +9,71 @@
 #
 ##############################################################################
 
-from odoo import api, models
+import logging
+
+from odoo import api, fields, models
+from odoo.tools.translate import _
+from odoo.exceptions import UserError
+
+
+_logger = logging.getLogger(__name__)
 
 
 class PortalWizard(models.TransientModel):
     _inherit = 'portal.wizard'
 
+    invitation_config_id = fields.Many2one('partner.communication.config')
+
+    @api.onchange('portal_id')
+    def onchange_portal_id(self):
+        # set the values of the users created in the super method
+        res = super(PortalWizard, self).onchange_portal_id()
+
+        self.user_ids.write({
+            'invitation_config_id': self.invitation_config_id.id
+        })
+
+        return res
+
+
+class PortalWizardUser(models.TransientModel):
+    _inherit = 'portal.wizard.user'
+
+    invitation_config_id = fields.Many2one('partner.communication.config')
+    uid_communication_id = fields.Many2one('partner.communication.job')
+
     @api.multi
     def action_apply(self):
-        """
-        Generate communication for User invitation
-        """
-        res = super(PortalWizard, self).action_apply()
-        comm_obj = self.env['partner.communication.job']
-        config = self.env.ref(
-            'partner_communication_switzerland.portal_welcome_config')
-        for user in self.mapped('user_ids.user_id'):
-            comm_obj.create({
-                'partner_id': user.partner_id.id,
-                'object_ids': user.id,
-                'config_id': config.id
-            })
+        res = super(PortalWizardUser, self).action_apply()
+
+        self.mapped('partner_id').signup_prepare()
+
+        if self.env.context['create_communication']:
+            for wizard_line in self:
+                wizard_line.create_uid_communication()
+        else:
+            for wizard_line in self:
```

Comment:
Why did you add this? I'm not sure we want to automatically activate the account if we don't send a communication.

Reply:
I was afraid to give to much work to the user to activate manually the account, because if you dont send a email with a link to activate the account, who gonna activate it ?

Reply:
In fact we register some users that we never want to give access to the account, only to be able to assign them to some events or other objects (but this is intern stuff and we don't communicate to them). So we don't wan't that everyone is activated automatically. I think you can remove this part.

Reply:
alright, it's gonna be done